reTurn/client/TurnAsyncSocket.cxx

```diff
@@ -1114,6 +1115,33 @@ TurnAsyncSocket::sendToRemotePeer(RemotePeer& remotePeer, const std::shared_ptr<
    }
 }
 
+void
+TurnAsyncSocket::setSoftware(const std::string &software)
+{
+   mSoftware = software;
+
+   const size_t unpadded = mSoftware.size();
+
+   if (unpadded > 0)
+   {
+      // Pad size to a multiple of 4, to help compatibility with older clients
+      const size_t remainder = unpadded % 4,
+                   padded    = remainder ? unpadded + 4 - remainder : unpadded;
+
+      if (padded > unpadded)
+         mSoftware.append(padded - unpadded, ' ');
+   }
+}
+
+const char *
+TurnAsyncSocket::getSoftware() const
+{
+   if (mSoftware.empty())
```

Comment:
This if seems a little odd, since you initialize mSoftware to SOFTWARE_STRING in the constructor.  Perhaps this is just protection incase someone set's to an empty string?  I think the logic should be different if someone set's it to an empty string though - I think we should not even add the software attribute in that case.  It is an optional attribute.